### PR TITLE
Todolists add remove

### PR DESCRIPTION
**Pull request by haftamudesta**
This project is a to-do list. A to-do list is a tool that helps to organize your day. It simply lists the things that you need to do and allows you to mark them as complete. this project allows you to add, remove edit, and display lists. It is implemented with HTML, CSS, and JavaScript.

### CODE REVIEW

**Comment by julie-ify** [comment]
Hi @haftamudesta,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly update your readme file using [the template](https://github.com/microverseinc/readme-template)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: module/addRemove.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  editing = todoId;
+};
+
+const deleteTodo = (todoId) => {
+  todos.filter((todo, index) => index !== todoId);
+  editing = -1;
+  todos.length -= 1;
+  displayTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
```
Comment (ShadyShawkat) [inline module/addRemove.js]: - [ ] Kindly make sure that when the user removes a task, the task that the user clicked on gets removed and not any other task. You can check this in the below GIF (might take some time to load)

chrome-capture-2023-0-25

- [ ] Please make sure that removing a task updates all remaining items' indexes, as per [this](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Deleting%20a%20task%20should%20update%20all%20remaining%20items%27%20indexes%2C%20so%20they%20represent%20the%20current%20list%20order%20and%20are%20unique(i.e.%20if%20you%27re%20deleting%20the%20first%20task%20index%201%20from%20the%20list%2C%20the%20index%20of%20the%20next%20task(2)%20should%20set%20to%201)..) requirement.

File: module/addRemove.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        value: todo.value,
+        color: todo.color,
+        checked: todo.checked,
+        index: index,
+      };
+    }
+  });
+  displayTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
```
Comment (ShadyShawkat) [inline module/addRemove.js]: - [ ] Please make sure that the index gets stored in the local storage, as per [this](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=By%20default%20new%20tasks%20should%20have%20the%20property%20completed%20set%20to%20false%20and%20the%20property%20index) requirement.
image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: module/addRemove.js (hunk 1 above)
Comment (Whoistolu) [reply module/addRemove.js]: - Well resolved @haftamudesta 👍🏽

File: module/addRemove.js (hunk 2 above)
Comment (Whoistolu) [reply module/addRemove.js]: - [ ] [REQUIRED] Well done 👍 I can now see the index in the Local storage, however, there is still one core requirement you are missing out here. The property index should be set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal 5). The first index starting from 0 should start from 1 not 0.
- You can fix this by adding 1 to your index.
### Here is it from the project requirements:
Screenshot 2023-01-27 at 13 38 39

### Here is what we currently have in your Local storage
Screenshot 2023-01-27 at 13 27 10

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @haftamudesta 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi @haftamudesta , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights'

- All linters are passing ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**